pytorch_lightning/trainer/trainer.py

```diff
@@ -820,91 +824,14 @@ def slurm_job_id(self) -> int:
-        parser.add_argument('--num_sanity_val_steps', default=5, type=int,
-                            help='check runs n batches of val before starting the training')
-        parser.add_argument('--process_position', default=0, type=int,
-                            help='orders the tqdm bar')
-        parser.add_argument('--show_progress_bar', default=True, type=bool,
-                            help='If true shows tqdm progress bar')
-        parser.add_argument('--distributed_backend', default=None, type=Optional[str],
-                            help='The distributed backend to use')
-        parser.add_argument('--weights_summary', default='full', type=str,
-                            help='Prints a summary of the weights when training begins')
-        parser.add_argument('--profiler', default=None, type=Optional[BaseProfiler],
-                            help='To profile individual steps during training and assist in'
-                                 'identifying bottlenecks')
-        parser.add_argument('--precision', default=32, type=int)
-
-        # model path
-        parser.add_argument('--default_save_path', default=None, type=Optional[str],
-                            help='Default path for logs and weights')
-        parser.add_argument('--weights_save_path', default=None, type=Optional[str],
-                            help='Prints a summary of the weights when training begins')
-
-        # GPU
-        parser.add_argument('--gpus', default=None, type=Optional[Union[List[int], str, int]])
-        parser.add_argument('--num_nodes', dest='num_nodes', type=int, default=1)
-        parser.add_argument('--num_tpu_cores', default=None, type=Optional[int])
-        parser.add_argument('--use_amp', dest='use_amp', default=False, type=bool)
-        parser.add_argument('--amp_level', dest='amp_level', default='O1', type=str,
-                            help='the optimization level to use')
-
-        # Fast Training
-        parser.add_argument('--fast_dev_run', dest='fast_dev_run', default=False, type=bool,
-                            help='runs validation after 1 training step')
-        parser.add_argument('--overfit_pct', default=0.0, type=float, dest='overfit_pct',
-                            help='%% of dataset to use with this option. float, or -1 for none')
-
-        # log
-        parser.add_argument('--logger', default=True, type=Union[LightningLoggerBase, bool])
-        parser.add_argument('--log_gpu_memory', default=None, type=Optional[str])
-        parser.add_argument('--row_log_interval', default=10, type=int,
-                            help='add log every k batches')
-        parser.add_argument('--log_save_interval', default=100, type=int,
-                            help='how many batches between log saves')
+        trainer_default_args = vars(Trainer())
+
+        for arg in trainer_default_args:
+            parser.add_argument('--{0}'.format(arg), default=trainer_args[arg], dest=arg)
```

Comment:
is there a way how to get also help texts?

Reply:
looking into it